pxr/imaging/plugin/hdRpr/light.cpp

```diff
@@ -385,43 +390,36 @@ void HdRprLight::Sync(HdSceneDelegate* sceneDelegate,
 
             }
 
-            bool operator()(LightVariantEmpty) const { return false; }
-            bool operator()(AreaLight* light) const {
+            void operator()(LightVariantEmpty) const { /*no-op*/ }
+            void operator()(AreaLight* light) const {
                 if (emissionColorIsDirty || !light->material) {
-                    MaterialAdapter matAdapter(EMaterialType::EMISSIVE, MaterialParams{{HdLightTokens->color, VtValue(emissionColor)}});
-                    light->material = std::move(rprApi->CreateMaterial(matAdapter));
+                    if (light->material) {
+                        rprApi->ReleaseGeometryLightMaterial(light->material);
+                    }
+                    light->material = rprApi->CreateGeometryLightMaterial(emissionColor);
                 }
 
                 if (light->material) {
                     for (auto& mesh : light->meshes) {
                         rprApi->SetMeshMaterial(mesh, light->material, false, false);
                     }
-                    return true;
                 }
-
-                return false;
             }
 
-            bool operator()(rpr::SpotLight* light) const {
+            void operator()(rpr::SpotLight* light) const {
                 if (emissionColorIsDirty) { rprApi->SetLightColor(light, emissionColor); }
-                return true;
             }
 
-            bool operator()(rpr::PointLight* light) const {
+            void operator()(rpr::PointLight* light) const {
                 if (emissionColorIsDirty) { rprApi->SetLightColor(light, emissionColor); }
-                return true;
             }
 
-            bool operator()(rpr::IESLight* light) const {
+            void operator()(rpr::IESLight* light) const {
                 if (emissionColorIsDirty) { rprApi->SetLightColor(light, emissionColor); }
-                return true;
             }
         };
 
-        if (BOOST_NS::apply_visitor(LightParameterSetter{rprApi, emissionColor, isEmissionColorDirty}, m_light) && !m_created) {
-            m_created = true;
-            rprRenderParam->AddLight();
-        }
+        BOOST_NS::apply_visitor(LightParameterSetter{rprApi, emissionColor, isEmissionColorDirty}, m_light);
```

Comment:
Anyway we can get away with not using boost libraries here?

Reply:
This exact block can be replaced with a wall of if-else:
```
if (m_light.which() == kLightTypePoint) {
    // point light logic
} else if (...) {
    ...
}
```

But it does not make sense. If we want to get rid of Boost then we need to remove it everywhere. Do you have any information about Boost in USD in the feature? Right now it's one of the mandatory requirements for building